networking_aci/plugins/ml2/drivers/mech_aci/driver.py

```diff
@@ -79,39 +76,22 @@ def bind_port(self, context):
         host_id, host_config = self._host_or_host_group(host)
 
         if not host_config:
-            return False
+            return
 
         segment_type = host_config.get('segment_type', 'vlan')
         segment_physnet = host_config.get('physical_network', None)
 
         for segment in context.segments_to_bind:
-            if context.binding_levels is None:
-                # For now we assume only two levels in hierarchy. The top level VXLAN/VLAN and
-                # one dynamically allocated segment at level 1
-                level = 1
-                allocation = self.allocations_manager.allocate_segment(network, host_id, level, host_config)
-
-                if not allocation:
-                    LOG.error("Binding failed, could not allocate a segment for further binding levels "
-                              "for port %(port)s",
-                              {'port': context.current['id']})
-                    return False
-
-                next_segment = {
-                    'segmentation_id': allocation.segmentation_id,
-                    'network_id': network_id,
-                    'network_type': segment_type,
-                    'physical_network': segment_physnet,
-                    'id': allocation.segment_id,
-                    'is_dynamic': False,
-                    'segment_index': level
-                }
+            if segment[api.NETWORK_TYPE] == lib_const.TYPE_VXLAN:
```

Comment:
segment_index=0 is something that's purely used by aci driver for it's own reference. Level 0 is already ensured since the segments_to_bind contains an ordered list of the segments. I think in rare cases we had 0 for vlans.

I am a bit in favor about dropping segment_index completely since it seems a source of confusion:
Networking-ACI has the vice-versa definition of low and high segments. In OpenStack, there are lower and higher segments in HPB:
https://specs.openstack.org/openstack/neutron-specs/specs/kilo/ml2-hierarchical-port-binding.html
Lowest = segment level 0 = compute_nodes, filers, loadbalancers and so on, thereas higher levels (up to 10) are transitioning network switches.

not sure what's the best way to go